nicsdru_nidirect_theme.theme

```diff
@@ -232,7 +233,13 @@ function _build_audit_link($type, &$variables) {
     // Get the current node.
     $node = Drupal::routeMatch()->getParameter('node');
     if (!empty($node)) {
-      $nid = $node->id();
+      $nid = NULL;
+      if (is_object($node)) {
```

Comment:
I'd check against NodeInterface, you could still have an object there that doesn't have an id method.

Reply:
Yep, same observation here. It's a technique used extensively in core modules and saves a few extra conditional steps along the way.

Reply:
Yes, great idea, I just did a sweep through core and have lifted code to check more carefully